worksheet_C/readme.MD

```diff
@@ -0,0 +1,31 @@
+##Question A
+This algorith checks both lists for duplicate entries.
+
+##Question B
+The list is being checked for duplicates which means each number in one list must be checked against every single number from the other list, essentially in pairs which is why quadratic would be the worst case.
+
+##Question C
+The algorith would still be correct because the original was setup to check every number in each list, which would have meant two items being checked twice just in opposite. An example is listed below:
+
+A:B
+A:A
+B:B
+B:A
+
+However I also think it worth noting that this question was worded in such a way that there could be two answers.
+
+##Question D
+The new algorithm essentially removes one entry to be checked which might not look like a lot, as this means that there is one less number that has to be checked against every other entry in the list.
```

Comment:
Your answer to part (c) pretty much explains why the algorithm is twice as fast (the original algorithm checks each pair twice whereas this one checks only once). Not sure what you mean here though.